subprojects/jacoco/src/main/java/org/gradle/internal/jacoco/JacocoAgentJar.java

```diff
@@ -62,15 +49,7 @@ public void setAgentConf(FileCollection agentConf) {
      * @return a file pointing to the {@code jacocoagent.jar}
      */
     public File getJar() {
-        if (agentJar == null) {
-            agentJar = fileOperations.zipTree(getAgentConf().getSingleFile()).filter(new Spec<File>() {
-                @Override
-                public boolean isSatisfiedBy(File file) {
-                    return file.getName().equals("jacocoagent.jar");
-                }
-            }).getSingleFile();
-        }
-        return agentJar;
+        return getAgentConf().getSingleFile();
```

Comment:
:x: There are tests failing on CI. Am not sure what's going on there (wild guess: maybe this needs to be synchronized?).